### ISSUE TITLE
jvm crash problems with FrameName::javaMethodName 

### ISSUE THREAD

**Issue by iamcaoxudong** [jvm-profiling-tools/async-profiler#441]
1. Async-profiler version
async-profiler-2.0-linux-x64

2. What I did
The HDFS NameNode process is profiled in an loop for 30s each time as below. After several successful executions,  JVM crashed.
  ```
  for((i=0;i<1000;i++))
  do
  time=$(date +%Y-%m-%d_%H-%M-%S.%N | cut -b 1-23)
  ./profiler.sh -d 30 -f profiles/profile_$time.html 82188
  done
  ```

3. JVM version and stacks
please see the attachment hs_err_pid file for details

[hs_err_pid82188.log](https://github.com/jvm-profiling-tools/async-profiler/files/6627789/hs_err_pid82188.log)

**Comment by apangin** [comment]
The crashlog tells that there are 3 instances of async-profiler loaded in the JVM process:
```
/tmp/async-profiler-2.0-linux-x64/build/libasyncProfiler.so (deleted)
/data1/async-profiler-2.0-linux-x64/build/libasyncProfiler.so
/data/async-profiler/build/libasyncProfiler.so
```
Furthermore, there are many `anon_inode:[perf_event]` mappings active.

At the moment of crash, the profiler is dumping collected data. At this point, the profiler must be stopped, and there should be no active perf_events. So, I can conclude that multiple profiler instances run simultaneously and conflict with each other.

Currently there is no protection against running multiple copies of the profiler. I'll think how to improve this in future. As for now, just make sure only one instance of the profiler is used.

**Comment by iamcaoxudong** [comment]
Many thanks for your explanation! 

By the way, the async-profiler is so powerful that we use it very often, and there may be some weird uses, will any of the following usages cause JVM crash possibly?
1. two simultaneous profiling procedures with identical libasyncProfiler.so (e.g. all use  /tmp/async-profiler-2.0-linux-x64)
2. two simultaneous profiling procedures with two libasyncProfiler.so of same version but under different paths （e.g. /tmp/async-profiler-2.0-linux-x64 and /data1/async-profiler-2.0-linux-x64）
3. two simultaneous profiling procedures with two libasyncProfiler.so of different version as well as under different paths （e.g. /tmp/async-profiler-2.0-linux-x64 and /data/async-profiler/).

**Comment by apangin** [comment]
Let me first tell why simultaneous profiling sessions are not supported.
async-profiler relies on the OS signals and on the instrumentation of JVM functions. In both cases, this implies a process-wide effect, i.e. two sessions can't be isolated from each other.

When you attempt to run the profiler twice using the same binary on the same path, you'll simply get `Profiler already started` error. There is a protection on the shared library level; JVM should not crash in this case.

However, when you run two instances of the profiler on different paths, no matter if they are of the same version or not, these will be two different shared libraries that don't know about each other. In this case they may conflict with unpredictable consequences, up to the JVM crash.

**Comment by iamcaoxudong** [comment]
OK, got it, thanks very much!

**Comment by cloud-66** [comment]
I also have crashed jvm when i used v2.0 profiler . It was a production.
I can't give the full log, but some part of it is below

`#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fdfd9b6076f, pid=22656, tid=0x00007fdfdcc01700
#
# JRE version: OpenJDK Runtime Environment (8.0_275-b01) (build 1.8.0_275-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.275-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libasyncProfiler.so+0x2876f]  Profiler::dumpFlameGraph(std::ostream&, Arguments&, bool)+0x1ff
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   https://bugzilla.redhat.com/enter_bug.cgi?product=Red%20Hat%20Enterprise%20Linux%207&component=java-1.8.0-openjdk
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  T H R E A D  ---------------

Current thread (0x0000556e46e98000):  JavaThread "Attach Listener" daemon [_thread_in_native, id=25681, stack(0x00007fdfdc701000,0x00007fdfdcc02000)]

siginfo: si_signo: 11 (SIGSEGV), si_code: 1 (SEGV_MAPERR), si_addr: 0x0000000000000000

Registers:
RAX=0x000000000000013b, RBX=0x0000556e4a3e3938, RCX=0x0000000000000000, RDX=0x0000000000000000
RSP=0x00007fdfdcbfddb0, RBP=0x00007fdfdcbff4a0, RSI=0x00007fdfdcbff740, RDI=0x00007fdfd9d77aa0
R8 =0x0000000000000000, R9 =0x0000556e460f0380, R10=0x0000556e460f0398, R11=0x0000000000000050
R12=0x0000556e4a3e3910, R13=0x0000556e4a3e3910, R14=0x0000000000000000, R15=0x00007fdfdcbff3c0
RIP=0x00007fdfd9b6076f, EFLAGS=0x0000000000010246, CSGSFS=0x0000000000000033, ERR=0x0000000000000004
  TRAPNO=0x000000000000000e

Top of Stack: (sp=0x00007fdfdcbfddb0)
0x00007fdfdcbfddb0:   00007fdfdcbfde10 00007fdfdb7d1a40
0x00007fdfdcbfddc0:   00007fdfdcbfee80 00007fdfdcbff450
0x00007fdfdcbfddd0:   00000006ce738178 00007fdfdcbff740
0x00007fdfdcbfdde0:   00007fdfdcbff500 00007fdfd9d77aa0
0x00007fdfdcbfddf0:   0000000000000000 000000000000013b
0x00007fdfdcbfde00:   000000073f1632c0 0000556e490d67e0
0x00007fdfdcbfde10:   0000556e46e98000 0000000000000000
0x00007fdfdcbfde20:   0000556e46fdbae0 0000556e46fdf1b0
0x00007fdfdcbfde30:   0000556e464e23c0 0000000000000005
0x00007fdfdcbfde40:   0000000000000080 0000000000000000
0x00007fdfdcbfde50:   3533342c31312866 7573272c302c312c
0x00007fdfdcbfde60:   552f6373696d2f6e 6e752e656661736e
0x00007fdfdcbfde70:   000a29276b726170 7075536b636f4c2f
0x00007fdfdcbfde80:   706e752e74726f70 7a000a29276b7261
0x00007fdfdcbfde90:   7261706e752e7265 737365636375536b
0x00007fdfdcbfdea0:   6b63000a2927726f 7075727265746e49
0x00007fdfdcbfdeb0:   6157656c69685774 0a2927676e697469
0x00007fdfdcbfdec0:   63246164000a2900 2730246574616572
0x00007fdfdcbfded0:   632e333534000a29 00000a29276c6c61
0x00007fdfdcbfdee0:   00007fdfdcbfdf30 00007fdffc4eb0ce
0x00007fdfdcbfdef0:   00007fdfdcbfe1b0 00000007719b9108
0x00007fdfdcbfdf00:   0000000000000000 00007fdffd0703fc
0x00007fdfdcbfdf10:   00007fdfdcbfdf50 00007fe00d62238c
0x00007fdfdcbfdf20:   00007fdfdcbfe540 00007fe00de9c080
0x00007fdfdcbfdf30:   00007fdfdcbfe6e0 00007fe00d62238a
0x00007fdfdcbfdf40:   00007fdfdcbfe550 00007fe00db2073b
0x00007fdfdcbfdf50:   00000007719b9148 00000006c0a116a8
0x00007fdfdcbfdf60:   0000556e46e98000 0000000700000001
0x00007fdfdcbfdf70:   0000556e46e98000 00007fe0d8142308
0x00007fdfdcbfdf80:   00000006c0a11580 00000007c0016c80
0x00007fdfdcbfdf90:   00007fdf00000029 0000000000030000
0x00007fdfdcbfdfa0:   0000000000000000 00007fdfdcbfe080 

Instructions: (pc=0x00007fdfd9b6076f)
0x00007fdfd9b6074f:   48 8b 40 08 48 89 85 58 e9 ff ff 48 8b b5 38 e9
0x00007fdfd9b6075f:   ff ff 48 8b 8d 50 e9 ff ff 80 be a0 00 00 00 00
0x00007fdfd9b6076f:   8b 09 89 8d 64 e9 ff ff 0f 84 ae 03 00 00 48 8b
0x00007fdfd9b6077f:   85 48 e9 ff ff 48 8b 9d 10 e9 ff ff 80 b8 84 82 

Register to memory mapping:

RAX=0x000000000000013b is an unknown value
RBX=0x0000556e4a3e3938 is an unknown value
RCX=0x0000000000000000 is an unknown value
RDX=0x0000000000000000 is an unknown value
RSP=0x00007fdfdcbfddb0 is pointing into the stack for thread: 0x0000556e46e98000
RBP=0x00007fdfdcbff4a0 is pointing into the stack for thread: 0x0000556e46e98000
RSI=0x00007fdfdcbff740 is pointing into the stack for thread: 0x0000556e46e98000
RDI=0x00007fdfd9d77aa0: <offset 0x23faa0> in /tmp/stack/async-profiler/build/libasyncProfiler.so at 0x00007fdfd9b38000
R8 =0x0000000000000000 is an unknown value
R9 =0x0000556e460f0380 is an unknown value
R10=0x0000556e460f0398 is an unknown value
R11=0x0000000000000050 is an unknown value
R12=0x0000556e4a3e3910 is an unknown value
R13=0x0000556e4a3e3910 is an unknown value
R14=0x0000000000000000 is an unknown value
R15=0x00007fdfdcbff3c0 is pointing into the stack for thread: 0x0000556e46e98000


Stack: [0x00007fdfdc701000,0x00007fdfdcc02000],  sp=0x00007fdfdcbfddb0,  free space=5107k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libasyncProfiler.so+0x2876f]  Profiler::dumpFlameGraph(std::ostream&, Arguments&, bool)+0x1ff
C  [libasyncProfiler.so+0x2a77f]  Profiler::runInternal(Arguments&, std::ostream&)+0x38f
C  [libasyncProfiler.so+0x2aa22]  Profiler::run(Arguments&)+0x282
C  [libasyncProfiler.so+0x31c8f]  Agent_OnAttach+0x18f
V  [libjvm.so+0x7b98dd]  JvmtiExport::load_agent_library(AttachOperation*, outputStream*)+0x31d
V  [libjvm.so+0x33309f]  attach_listener_thread_entry(JavaThread*, Thread*)+0x14f
V  [libjvm.so+0xb032f2]  JavaThread::thread_main_inner()+0x212
V  [libjvm.so+0x93c7e2]  java_start(Thread*)+0xf2
C  [libpthread.so.0+0x7ea5]  start_thread+0xc5`

alse there are exceptions in the log

`Internal exceptions (250 events):
Event: 1130389.492 Thread 0x0000556e47b3b800 Exception <a 'java/net/SocketException': Socket closed> (0x000000076db660f8) thrown at [/builddir/build/BUILD/java-1.8.0-openjdk-1.8.0.275.b01-0.el7_9.x86_64/openjdk/hotspot/src/share/vm/prims/jni.cpp, line 738]
Event: 1130389.504 Thread 0x0000556e48422800 Exception <a 'java/net/SocketTimeoutException': Read timed out> (0x000000076da8f1b0) thrown at [/builddir/build/BUILD/java-1.8.0-openjdk-1.8.0.275.b01-0.el7_9.x86_64/openjdk/hotspot/src/share/vm/prims/jni.cpp, line 738]
Event: 1130389.586 Thread 0x0000556e47b3b800 Exception <a 'java/net/SocketException': Socket closed> (0x000000076dd26430) thrown at [/builddir/build/BUILD/java-1.8.0-openjdk-1.8.0.275.b01-0.el7_9.x86_64/openjdk/hotspot/src/share/vm/prims/jni.cpp, line 738]`

**Comment by apangin** [comment]
@cloud-66 Please open a new issue. This question is already answered, and your crash does not look related.
